engine/classes/Elgg/I18n/Translator.php

```diff
@@ -42,6 +42,13 @@ function translate($message_key, $args = array(), $language = "") {
 	
 		static $CURRENT_LANGUAGE;
 	
+		if (!is_string($message_key) || strlen($message_key) < 1) {
```

Comment:
i needed to validate it's a string and technically '0' is still valid but empty('0') is true